File: resolucoes/kotlin/lista01/src/main/kotlin/ex006.kt
import java.util.Scanner

/**
 * Faça um programa que leia um número inteiro e mostre o seu antecessor e seu
 * sucessor.
 * Ex:
 * Digite um número: 9
 * O antecessor de 9 é 8
 * O sucessor de 9 é 10
 * */
fun ex006() {
    val scanner = Scanner(System.`in`)
    print("Digite um número: ")
    val number = scanner.nextInt()
    println(
        """
        O antecessor de $number é ${number - 1}
        O sucessor de $number é ${number + 1}
    """.trimIndent()
    )
}

internal fun main() {
    ex006()
}

File: resolucoes/TypeScript/lista01/ex006.ts
import prompt from "prompt-sync"

export function main() {
    const p = prompt()

const int = Number(p("Digite um número inteiro:"))
const ant = int - 1
const suc = int + 1
p(`Seu número inteiro é ${int} o antecessor é ${ant} e seu sucessor é ${suc}`)
}